sources/standard/inbox/__init__.py

```diff
@@ -0,0 +1,247 @@
+    """
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        for item in items:
+            message_uid = str(item["message_uid"])
+            msg = get_message(client, message_uid)
+            if msg:
+                result = deepcopy(item)
+                result["modification_date"] = get_internal_date(client, message_uid)
+                result.update(extract_email_info(msg, include_body=include_body))
+
+                yield result
+
+
+class ImapFileModel(FileItem):
+    """A DataItem representing an email attachment"""
+
+    data_hash: str
+
+
+@dlt.transformer(
+    name="attachments",
+    write_disposition="merge",
+    merge_key="data_hash",
+    primary_key="data_hash",
+)
+def get_attachments_by_uid(
+    items: TDataItems,
+    storage_path: str,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    filter_by_mime_type: Sequence[str] = (),
+) -> TDataItem:
+    """Downloads attachments from email messages based on the provided message UIDs.
+
+    Args:
+        items (TDataItems): An iterable containing dictionaries with 'message_uid' representing the email message UIDs.
+        storage_path (str): The local folder path where attachments will be downloaded.
+        host (str, optional): The hostname of the IMAP server. Default is 'dlt.secrets.value'.
+        email_account (str, optional): The email account used to log in to the IMAP server. Default is 'dlt.secrets.value'.
+        password (str, optional): The password for the email account. Default is 'dlt.secrets.value'.
+        filter_by_mime_type (Sequence[str], optional): A sequence of MIME types used to filter attachments based on their content type. Default is an empty sequence.
+
+    Yields:
+        TDataItem: A dictionary containing the collected email information and attachment details.
+    """
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        for item in items:
+            message_uid = str(item["message_uid"])
+            msg = get_message(client, message_uid)
+            if isinstance(filter_by_mime_type, str):
+                filter_by_mime_type = [filter_by_mime_type]
+            if filter_by_mime_type:
+                attachments = None
+                for mime_type in filter_by_mime_type:
+                    new_attachments = extract_attachments(msg, mime_type)
+                    if not attachments:
+                        attachments = new_attachments
+                    else:
+                        attachments = chain(attachments, new_attachments)
+            email_info = extract_email_info(msg)
+            internal_date = get_internal_date(client, message_uid)
+
+            for attachment in attachments:
+                filename = f"{message_uid}_{attachment['file_name']}"
```

Comment:
my take:
1. remove copying file to local
2. load the bytes into file_content
3. return FileSystemDict as in filesystem